cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java

```diff
@@ -70,7 +70,13 @@ public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreExcep
     createLocation(namespaceMeta);
 
     if (cConf.getBoolean(Constants.Explore.EXPLORE_ENABLED)) {
-      exploreFacade.createNamespace(namespaceMeta.getNamespaceId().toId());
+      try {
+        exploreFacade.createNamespace(namespaceMeta.getNamespaceId().toId());
+      } catch (ExploreException | SQLException e) {
+        // if we failed to create a namespace in explore then delete the earlier created location for the namespace
+        deleteLocation(namespaceMeta.getNamespaceId());
```

Comment:
this is not fool-proof right? I mean what about the hbase namespace, if it was created? 

Shouldn't the proper way be that in ``DefaultNamespaceAdmin``, we clean things up (via storageprovidernamespaceadmin.delete) in a finally?